### ISSUE TITLE
kubectl test adds taints to a node and may interfere with other test

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#36781]
Forked from https://github.com/kubernetes/kubernetes/issues/35197#issuecomment-260493189

The kubectl test "should remove all the taints with the same key off a node" adds taint to the node, which may interfere with other tests. 

E.g., https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gke/3881/
```
[BeforeEach] [k8s.io] Kubectl client
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:282
[It] should remove all the taints with the same key off a node
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:1434
[1mSTEP[0m: adding the taint kubernetes.io/e2e-taint-key-83fc44ac-a9f4-11e6-b3d6-0242ac110003=testing-taint-value:NoSchedule to a node
Nov 13 14:57:13.287: INFO: Running '/workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://104.198.70.74 --kubeconfig=/workspace/.kube/config taint nodes gke-jenkins-e2e-default-pool-a4eebde8-om8a kubernetes.io/e2e-taint-key-83fc44ac-a9f4-11e6-b3d6-0242ac110003=testing-taint-value:NoSchedule'
Nov 13 14:57:13.552: INFO: stderr: ""
Nov 13 14:57:13.552: INFO: stdout: "node \"gke-jenkins-e2e-default-pool-a4eebde8-om8a\" tainted\n"
[1mSTEP[0m: verifying the node has the taint kubernetes.io/e2e-taint-key-83fc44ac-a9f4-11e6-b3d6-0242ac110003=testing-taint-value:NoSchedule
Nov 13 14:57:13.552: INFO: Running '/workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://104.198.70.74 --kubeconfig=/workspace/.kube/config describe node gke-jenkins-e2e-default-pool-a4eebde8-om8a'
```

The node `gke-jenkins-e2e-default-pool-a4eebde8-om8a` was considered unschedulable in a non-related test and caused confusion/failures.

/cc @kubernetes/kubectl

**Comment by bprashanth** [comment]
Probably just run them in serial, given https://github.com/kubernetes/kubernetes/issues/31906 @kevin-wangzefeng

**Comment by yujuhong** [comment]
Assigned to @kevin-wangzefeng and @davidopp (owner of #31906).

**Comment by xiangpengzhao** [comment]
and
/sig scheduling

**Comment by yujuhong** [comment]
Fixed by #36820
